cooperator_website/controllers/main.py

```diff
@@ -236,18 +237,23 @@ def fill_values(self, values, is_company, logged, load_from_user=False):
     def get_products_share(self, is_company):
         product_obj = request.env["product.template"]
         products = product_obj.sudo().get_web_share_products(is_company)
-
         return products
 
     def get_countries(self):
         countries = request.env["res.country"].sudo().search([])
-
         return countries
 
     def get_langs(self):
         langs = request.env["res.lang"].sudo().search([])
         return langs
 
+    def get_company_types(self):
+        company_types_records = (
+            request.env["res.partner.company.type"].sudo().search([])
+        )
+        company_types = [(record.id, record.name) for record in company_types_records]
```

Comment:
i think it would be a good addition to also display the abbreviation next to the name:

```suggestion
        company_types = [
            (
                record.id,
                "{name} ({abbr})".format(name=record.name, abbr=record.shortcut),
            )
            for record in company_types_records
        ]
```

Reply:
actually, i’m not so sure about this. maybe this should be changed directly in the `partner_company_type` module (for example, having a “display name” field), to ensure consistency (also in the internal partner forms, for example).